src/components/common/Text/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { ReactElement } from 'react';
+import { useTheme } from 'styled-components';
+
+import fonts from '@theme/fonts';
+import { IPalette } from '@theme/theme';
+import { typography } from '@theme/typography';
+
+type TextSX = {
+  fontSize?: keyof typeof fonts.fontSize;
+  fontWeight?: keyof typeof fonts.fontWeight;
+  fontFamily?: keyof typeof fonts.fontFamily;
+  fontStyle?: keyof typeof fonts.fontStyle;
+  color?: keyof IPalette;
+  lineHeight?: keyof typeof fonts.lineHeight;
+  letterSpacing?: keyof typeof fonts.letterSpacing;
+};
+
+type TextProps = {
+  variant?: keyof typeof typography;
+  sx?: TextSX;
+  render: (style) => ReactElement;
+};
+
+const Text = ({ variant, sx, render }: TextProps) => {
+  const theme = useTheme();
+  const getVariantStyle = () => variant && typography[variant];
+  const getCustomStyle = () => {
+    const customStyle = { ...sx };
+
+    Object.keys(customStyle).forEach((key) => {
+      const value = customStyle[key];
+
+      customStyle[key] =
+        key === 'color' ? theme.palette[value] : fonts[key][value];
+    });
```

Comment:
어차피 key와 value 모두 사용할 거라면 Object.entries를 사용해도 되지 않을까 하는 생각이 들었습니다!

```suggestion
    Object.entries(customStyle).forEach(([key, value]) => {
      customStyle[key] = 
        key === 'color' ? theme.palette[value] : fonts[key][value];
    })
```